src/WebJobs.Script.WebHost/Management/WebFunctionsManager.cs

```diff
@@ -250,13 +255,23 @@ private async Task<(bool, string)> InternalSyncTriggers(IEnumerable<JObject> tri
 
         internal IEnumerable<FunctionMetadata> GetFunctionsMetadata()
         {
-            return FunctionMetadataManager
-                .ReadFunctionsMetadata(FileUtility.EnumerateDirectories(_config.RootScriptPath), null, _workerConfigs, _logger, fileSystem: FileUtility.Instance);
+            // get functions metadata
+            var functionDirectories = FileUtility.EnumerateDirectories(_hostOptions.RootScriptPath);
+            IEnumerable<FunctionMetadata> functionsMetadata = FunctionMetadataManager.ReadFunctionsMetadata(functionDirectories, null, _workerConfigs, _logger, fileSystem: FileUtility.Instance);
+
+            // get proxies metadata
```

Comment:
Why weren't we returning proxy metadata previously here? Was that intentional or just an omission?

Reply:
Are you saying this was missing and all tests were passing? 

---
In reply to: [213801895](https://github.com/Azure/azure-functions-host/pull/3351#discussion_r213801895) [](ancestors = 213801895)

Reply:
I believe that is intentional. This API is used by the portal to get a list of functions and I'm not sure it will will deal with the extra proxy metadata well. (/cc @ahmelsayed )

We discussed this when we were making the updates for LA, and deferred on doing it.

Reply:
Wondering now if this wasn’t by design. Since proxies have their own UI I believe, maybe we don’t want to list them from this API?

Reply:
It was. See my comment above

Reply:
@ahmelsayed can comment on this as he originally wrote this to support the portal

Reply:
I'll change back so that the API by default doesn't return proxies, but I'll retain the capability by putting it behind a query param. This being a general API, I think it makes sense for it to be able to return metadata for ALL functions.

Reply:
👍